src/main/java/jenkins/plugins/logstash/persistence/LogzioDao.java

```diff
@@ -0,0 +1,202 @@
+package jenkins.plugins.logstash.persistence;
+
+import com.github.wnameless.json.flattener.JsonFlattener;
+import com.google.gson.JsonObject;
+import io.logz.sender.FormattedLogMessage;
+import io.logz.sender.HttpsRequestConfiguration;
+import io.logz.sender.HttpsSyncSender;
+import io.logz.sender.SenderStatusReporter;
+import io.logz.sender.exceptions.LogzioParameterErrorException;
+import io.logz.sender.exceptions.LogzioServerErrorException;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Logz.io Data Access Object.
+ *
+ * @author Ido Halevi
+ *
+ */
+
+public class LogzioDao extends AbstractLogstashIndexerDao {
+    private final String TYPE = "jenkins_logstash_plugin";
+    private String key;
+    private String host;
+    private LogzioHttpsClient httpsClient;
+
+    //primary constructor used by indexer factory
+    public LogzioDao(String host, String key){
+        this(null, host, key);
+    }
+
+    // Factored for unit testing
+    LogzioDao(LogzioHttpsClient factory, String host, String key) {
+        this.host = host;
+        this.key = key;
+        try{
+            this.httpsClient = factory == null ? new LogzioHttpsClient(key, host, TYPE) : factory;
+        }catch (LogzioParameterErrorException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    @Override
+    public void push(String data) throws IOException {
+        JSONObject jsonData = JSONObject.fromObject(data);
+        JSONArray logMessages = jsonData.getJSONArray("message");
+        if (!logMessages.isEmpty()) {
+            try{
+                for (Object logMsg : logMessages) {
+                    JsonObject logLine = createLogLine(jsonData, logMsg.toString());
+                    httpsClient.send(new FormattedLogMessage((logLine + "\n").getBytes(Charset.forName("UTF-8"))));
+                }
+                httpsClient.flush();
```

Comment:
just a suggestion: maybe do the flush() in finally block?

Reply:
flush() throws exception, it will require try-catch in the finally block.